packages/amplify-category-api/src/index.ts

```diff
@@ -156,15 +151,8 @@ export const initEnv = async (context: $TSContext): Promise<void> => {
   /**
    * Check team provider info to ensure it hasn't already been created for current env
    */
-  const currentEnv = amplify.getEnvInfo().envName;
-  const teamProviderInfo = stateManager.getTeamProviderInfo();
-  if (
-    teamProviderInfo[currentEnv][categories]
-    && teamProviderInfo[currentEnv][categories][category]
-    && teamProviderInfo[currentEnv][categories][category][resourceName]
-    && teamProviderInfo[currentEnv][categories][category][resourceName]
-    && teamProviderInfo[currentEnv][categories][category][resourceName][rdsRegion]
-  ) {
+  const resourceParamManager = (await ensureEnvParamManager()).instance.getResourceParamManager(category, resourceName);
+  if (resourceParamManager.getParam(rdsRegion)) {
```

Comment:
I think a better check here would be `(await ensureEnvParamManager()).instance.hasResourceParamManager(category, resourceName)` then get the param manager inside the if statement